proxy/create_container_interceptor.go

```diff
@@ -74,50 +78,104 @@ func (i *createContainerInterceptor) InterceptRequest(r *http.Request) error {
 	return nil
 }
 
-func (i *createContainerInterceptor) setWeaveWaitEntrypoint(container *docker.Config) error {
-	if len(container.Entrypoint) == 0 {
-		image, err := i.proxy.client.InspectImage(container.Image)
+func (i *createContainerInterceptor) setWeaveWaitEntrypoint(container jsonObject) error {
+	var entrypoint []string
+	entrypoint, err := container.StringArray("Entrypoint")
+	if err != nil {
+		err, ok := err.(*UnmarshalWrongTypeError)
+		if !ok {
+			return err
+		}
+
+		got, ok := err.Got.(string)
```

Comment:
We could handle it in `StringArray`. It would make it looser in what we accept for `Dns`, `Binds`, and `DnsSearch`, as well. Not sure whether it can happen for "Cmd" (It's never shown in the docs). Will check that.